src/extractcode/extract.py

```diff
@@ -97,7 +97,21 @@
 ExtractEvent = namedtuple('ExtractEvent', 'source target done warnings errors')
 
 
-def extract(location, kinds=extractcode.default_kinds, recurse=False):
+def extract(location, kinds=extractcode.default_kinds, recurse=False, replace_originals=False):
+    events = list(extract_files(location, kinds, recurse))
+    if replace_originals:
+        for xevent in reversed(events):
```

Comment:
Is going reverse needed here? e.g. essentially bottom up instead of top down?
I sense this is needed but I am not 100% sure.

Reply:
otherwise one would move `tmp.zip-extracted` before moving `tmp.zip-extracted/innerArchive.zip-extracted`

Reply:
OK: that what I thought fuzzily somehow!